### PR TITLE
speed up - sample_spatial_and_temporal_locations_for_examples

### LINKED ISSUE #304: Speed up `Manager.sample_spatial_and_temporal_locations_for_examples()` by splitting `shuffled_t0_datetimes` across multliple processes
but wait until we've implemented #305, as #305 is likely to be sufficient!

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

- use multiprocess to get spatial locations for examples

Fixes #304 

## How Has This Been Tested?

normal unittests, which do cover this function

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataset/manager.py [hunk 1]
```diff
@@ -245,18 +245,37 @@ def sample_spatial_and_temporal_locations_for_examples(
...
-        # But wait until we've implemented issue #305, as that is likely to be sufficient!
-        (
-            x_locations,
-            y_locations,
-        ) = self.data_source_which_defines_geospatial_locations.get_locations(shuffled_t0_datetimes)
+
+        with futures.ThreadPoolExecutor() as executor:
+            tasks = []
+
+            # loop over data sources
```
Comment (peterdudfield) [inline nowcasting_dataset/manager.py:252]: change comment

**Comment by peterdudfield** [comment]
Move dback to a draft as https://github.com/openclimatefix/nowcasting_dataset/pull/375 might be better

**Comment by JackKelly** [comment]
I've skim-read the code and this looks good but I agree, let's work on the assumption that #375 will be sufficient :slightly_smiling_face:  If #375 isn't sufficient then let's revisit parellising `Manager.sample_spatial_and_temporal_locations_for_examples()`.  Is that what you had in mind?

**Comment by peterdudfield** [comment]
> I've skim-read the code and this looks good but I agree, let's work on the assumption that #375 will be sufficient 🙂 If #375 isn't sufficient then let's revisit parellising `Manager.sample_spatial_and_temporal_locations_for_examples()`. Is that what you had in mind?

Yea, hoepfully #375 will be enough, if not can revisit this. 
Perhaps I could close this PR for now, so that its not confusing

**Comment by JackKelly** [comment]
> Perhaps I could close this PR for now, so that its not confusing

Sounds good!  (But thank you anyway for writing this PR; it does look great!)

**Pull request closed without merging**